app/services/billing_entities/update_applied_dunning_campaign_service.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module BillingEntities
+  class UpdateAppliedDunningCampaignService < BaseService
+    Result = BaseResult[:billing_entity]
+    def initialize(billing_entity:, dunning_campaign: nil)
+      @billing_entity = billing_entity
+      @dunning_campaign = dunning_campaign
+    end
+
+    def call
+      return unless billing_entity.applied_dunning_campaign != dunning_campaign
```

Comment:
should it fail if billing entity is nil?

Reply:
Added, good spot! 🙌